cli/cli_app.go

```diff
@@ -228,6 +228,15 @@ func runCommand(command string, terragruntOptions *options.TerragruntOptions) (f
 	if err := setRoleEnvironmentVariables(terragruntOptions, "", nil); err != nil {
 		return err
 	}
+
+	if (command == "apply" || command == "apply-all") && util.ListContainsElement(terragruntOptions.TerraformCliArgs, "-destroy") {
```

Comment:
Here is one of the important twist. `terraform destroy` is still supported, but it actually replace the call by `terraform apply -destroy`. However, because terragrunt implement a `destroy-all` command that behave differently than `apply-all -destroy` as we execute destroy on the stack in the reverse order compare to apply, we must handle the arguments to be sure that the behaviour is still the one we expect.